src/Middleware/HeaderPropagation/src/HeaderPropagationState.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.HeaderPropagation
+{
+    public class HeaderPropagationState
```

Comment:
Since this is a public type I'd like to consider different names for this. Open to suggestions, my ideas:

`HeaderPropagationValues`

`PropagatedHeaders`

`PropagatedHeaderValues`.

Reply:
I was tempted to use somethingHeaders (like `PropagatedHeaders` you suggested) but then `PropagatedHeaders.Headers` feel awkward. Renaming the `Dictionary` to `Values` is not ideal either as it has a `Values` property :)

Anyway I don't have a particular preference, what's your pick? and what should be the property name?

Reply:
I'd pick `HeaderPropagationValues` of those.

Reply:
I think `Headers` is fine for the name

Reply:
done